psi4/driver/procrouting/sapt/sapt_proc.py

```diff
@@ -107,36 +107,45 @@ def run_sapt_dft(name, **kwargs):
     core.IO.set_default_namespace('dimer')
     data = {}
 
-    if (core.get_global_option('SCF_TYPE') == 'DF'):
-        # core.set_global_option('DF_INTS_IO', 'LOAD')
+    if (core.get_global_option('SCF_TYPE') in ['DF', 'DISK_DF']):
+        # Save integrals
+        # We want to try to re-use itegrals for the dimer and monomer SCF's.
+        # If we are using Disk based DF (DISK_DF) then we can use the
+        # DF_INTS_IO option.  MemDF does not know about this option but setting
+        # it will be harmless there.  Handle the case of either (a) the default
+        # DF is used, or (b) the user selects DF, or (c) the user very
+        # specifically sets DISK_DF.
         core.set_global_option('DF_INTS_IO', 'SAVE')
 
     # Compute dimer wavefunction
     hf_wfn_dimer = None
     if do_delta_hf:
-        if (core.get_global_option('SCF_TYPE') == 'DF'):
+        if (core.get_global_option('SCF_TYPE') in ['DF', 'DISK_DF']):
             core.set_global_option('DF_INTS_IO', 'SAVE')
-
         core.timer_on("SAPT(DFT):Dimer SCF")
         hf_data = {}
+
+        # core.set_global_option("SAVE_JK", True)
+        core.set_local_option("SCF", "SAVE_JK", True)
         hf_wfn_dimer = scf_helper("SCF", molecule=sapt_dimer, banner="SAPT(DFT): delta HF Dimer", **kwargs)
         hf_data["HF DIMER"] = core.variable("CURRENT ENERGY")
         core.timer_off("SAPT(DFT):Dimer SCF")
 
         core.timer_on("SAPT(DFT):Monomer A SCF")
-        if (core.get_global_option('SCF_TYPE') == 'DF'):
-            core.IO.change_file_namespace(97, 'dimer', 'monomerA')
+        # if (core.get_global_option('SCF_TYPE') in ['DF', 'DISK_DF']):
```

Comment:
Could you explain why you're commenting out the namespace changes?

Reply:
When we were trying to make the DF_INTS_IO path for re-using integrals work, the namespace appeared to be causing an issue; however, now that we pass JK-objects directly, we no longer have this issue. As such, I have uncommented the change of namespace for the change still passes the test.